File: project_euler_code/ruby/4.rb
(0..0).each { |max| puts max if !(-999..-100).each { |a| (a*a < max)? break : (a..-100).each { |b| ((a*b < max) ? break : max = a*b) if (a*b).to_s == (a*b).to_s.reverse  } } }

File: project_euler_code/javascript/4.js
var max = 0;

for (var a = 999; a >= 100; a--) {
  for (var b = a-1; b >= 100; b--) {
    var c = String(a*b);
    if (c == c.split("").reverse().join("")) {
      if (parseInt(c) > max) {
        max = parseInt(c);
        console.log(c);
      }
    }
  }
}
